src/mmdata.h

```diff
@@ -276,29 +277,36 @@ struct nullPntrStruct {
      */
     long poolLoc;
     /*! 
-     * allocated size of the memory block containing the \a pntrString.
+     * allocated size of the memory block containing the \a pntrString,
+     * excluding any hidden administrative data.
      * Note: this is the number of bytes, not elements!  Fixed to the size of a
      * single void* instance. 
      */
     long allocSize;
     /*! 
-     * currently used size of the memory block containing the \a pntrString.
+     * currently used size of the memory block containing the \a pntrString,
+     * excluding any hidden administrative data.
      * Note: this is the number of bytes, not elements!  Fixed to the size of a
      * single pointer element.
      */
     long actualSize;
     /*! 
      * memory for a single void* instance, set and fixed to the null pointer.
+     * A null usually marks the end of the array.
      */
     pntrString nullElement; };
 /*!
- * \var g_PntrNull. Global instance of a memory block structured like a
- * \a pntrString, but fixed in size and containing always exactly one null
+ * \var g_PntrNull. Global singleton instance of a memory block structured like
```

Comment:
It's not a singleton, in the sense that pointer comparison does not reliably detect empty `pntrString`s. https://github.com/metamath/metamath-exe/blob/1639113fe3ec39ea0bc57946f4c409671837c520/src/mmdata.c#L2499